torch/csrc/onnx/onnx.h

```diff
@@ -284,4 +277,28 @@ class GraphProto : public MicroProto<onnx_GraphProto> {
   }
 };
 
+class ModelProto : public MicroProto<onnx_ModelProto> {
+private:
+  std::string producer_name;
+  std::string producer_version;
+  std::string domain;
+  std::string doc_string;
+  // graph is a static field in nanopb struct.
+  GraphProto graph;
+public:
+  ModelProto() : MicroProto(onnx_ModelProto_init_default) {
+    proto.has_ir_version = true;
+    proto.ir_version = onnx_Version_IR_VERSION;
+    proto.producer_name = string(&producer_name, "pytorch");
+    // TODO: stop hard-coding this
+    proto.producer_version = string(&producer_version, "0.2");
+    proto.domain = string(&domain, "com.facebook");
+  }
+  void set_model_version(int64_t i) { proto.has_model_version = true; proto.model_version = i; }
+  void set_doc_string(const std::string& s) { proto.doc_string = string(&doc_string, s); }
+  GraphProto* mutable_graph() { return &graph; }
+  // set_graph should be called after encoding graph.
+  void set_graph() { proto.has_graph = true; proto.graph = graph.proto; }
```

Comment:
And then this goes away, and we get an exact API compatibility with the Google Protobuf C++ interface :)